addon/model.js

```diff
@@ -17,20 +35,16 @@ export default class Model {
       const property = schema.properties[key];
       const type = property.type;
 
-      if (type.match(/string|number|boolean|date/i)) {
-        if (key.match(/_id$/)) {
-          attrs[name] = belongsTo({ async: true });
-        } else {
-          attrs[name] = attr(type);
-        }
-      } else if (type === 'array') {
-        if (key.match(/_ids$/)) {
-          attrs[pluralize(name)] = hasMany({ async: true });
-        } else {
-          attrs[name] = attr({ defaultValue: [] });
-        }
-      } else if (type === 'object') {
-        attrs[name] = attr({ defaultValue: {} });
+      if (Ember.isArray(type)) {
+        const [firstType] = type;
+
+        const [attr, value] = buildProperty(key, name, firstType);
+
+        attrs[attr] = value;
+      } else {
+        const [attr, value] = buildProperty(key, name, type);
```

Comment:
const 'attr' has already been declared.<br>'attr' is already defined.<br>const 'value' has already been declared.<br>'value' is already defined.